expfmt/text_parse.go

```diff
@@ -125,6 +126,15 @@ func (p *TextParser) reset(in io.Reader) {
 	} else {
 		p.buf.Reset(in)
 	}
+
+	// Some clients like to throw in UTF-8 BOM chars
+	maybeBOM, err := p.buf.Peek(3)
+	if len(maybeBOM) == 3 && maybeBOM[0] == 0xEF && maybeBOM[1] == 0xBB && maybeBOM[2] == 0xBF && err == nil {
+		// The common case dealing with funny Windows clients
+		log.Debug("Discarding BOM")
```

Comment:
Will do.

On Tue, May 31, 2016 at 6:32 AM, Björn Rabenstein <notifications@github.com>
wrote:

> In expfmt/text_parse.go
> <https://github.com/prometheus/common/pull/42#discussion_r65157940>:
>
> > @@ -125,6 +126,15 @@ func (p *TextParser) reset(in io.Reader) {
> >  	} else {
> >  		p.buf.Reset(in)
> >  	}
> > +
> > +	// Some clients like to throw in UTF-8 BOM chars
> > +	maybeBOM, err := p.buf.Peek(3)
> > +	if len(maybeBOM) == 3 && maybeBOM[0] == 0xEF && maybeBOM[1] == 0xBB && maybeBOM[2] == 0xBF && err == nil {
> > +		// The common case dealing with funny Windows clients
> > +		log.Debug("Discarding BOM")
>
> We don't use any logging anywhere in this package. It might come as a
> surprise to the (possibly very diverse) users, and it's pulling in another
> dependency. I think it's fine to silently swallow the BOMs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/common/pull/42/files/1e1e0c1c905808faedb252d7fbb96f35a43726f7#r65157940>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AStJnk1lrBxaqHlLQzUtcFti7h2pqeAYks5qHA5AgaJpZM4Iozt3>
> .
>